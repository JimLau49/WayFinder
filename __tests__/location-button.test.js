import React from "react";
import LocationButton from "../src/components/location-button/location-button.component";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

describe("LocationButton component", () => {
  it("should match snapshot", () => {
    const tree = renderer.create(<LocationButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should call LocationButton after button press", () => {
    const mockOnLocationButtonPress = jest.fn();
    const wrapper = shallow(
      <LocationButton onLocationButtonPress={mockOnLocationButtonPress} />
    );
    const button = wrapper.find("View TouchableOpacity");
    button.simulate("press");
    expect(mockOnLocationButtonPress).toHaveBeenCalledTimes(1);
  });
});
