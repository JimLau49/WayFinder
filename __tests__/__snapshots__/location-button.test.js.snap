// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationButton component should match snapshot 1`] = `
<View
  style={
    Object {
      "bottom": 125,
      "position": "absolute",
      "right": 10,
    }
  }
>
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "rgba(0,0,0,0.2)",
          "borderRadius": 50,
          "borderWidth": 1,
          "height": 60,
          "justifyContent": "center",
          "opacity": 1,
          "width": 60,
        }
      }
      testID="locationButton"
    >
      <Text />
    </View>
  </RNGestureHandlerButton>
</View>
`;
